src/nbl/asset/utils/CHLSLCompiler.cpp

```diff
@@ -61,29 +92,158 @@ CHLSLCompiler::~CHLSLCompiler()
+        if (s.length() >= 6) {
+            std::vector<std::wstring::iterator> underscorePositions = {};
+            auto it = std::find(s.begin(), s.end(), '_');
+            while (it != s.end()) {
+                underscorePositions.push_back(it);
+                it = std::find(it + 1, s.end(), '_');
+            }
+
+            if (underscorePositions.size() == 2) 
+            {   
+                stageStr = shaderStageOverrideFromPragma != IShader::ESS_UNKNOWN ? std::wstring(overrideStageStr) : std::wstring(s.begin(), underscorePositions[0]);
+                // Version
+                std::wstring majorVersionString, minorVersionString;
+                int size = underscorePositions.size();
+                auto secondLastUnderscore = underscorePositions[size - 2];
+                auto lastUnderscore = underscorePositions[size - 1];
+                majorVersionString = std::wstring(secondLastUnderscore + 1, lastUnderscore);
+                minorVersionString = std::wstring(lastUnderscore + 1, s.end());
+                try
+                {
+                    int major = std::stoi(majorVersionString);
+                    int minor = std::stoi(minorVersionString);
+                    if (major < MajorReqVersion || (major == MajorReqVersion && minor < MinorReqVersion))
+                    {
+                        // Overwrite the version 
+                        logger.log("Upgrading shader stage version number to %i %i", system::ILogger::ELL_DEBUG, MajorReqVersion, MinorReqVersion);
+                        arguments[foundShaderStageArgumentValueIdx] = stageStr + L"_" + std::to_wstring(MajorReqVersion) + L"_" + std::to_wstring(MinorReqVersion);
+                    }
+                    else
+                    {
+                        // keep the version as it was
+                        arguments[foundShaderStageArgumentValueIdx] = stageStr + L"_" + majorVersionString + L"_" + minorVersionString;
+                    }
+                    return;
+                }
+                catch (const std::invalid_argument& e)
+                {
+                    logger.log("Parsing shader version failed, invalid argument exception: %s", system::ILogger::ELL_ERROR, e.what());
+                }
+                catch (const std::out_of_range& e)
+                {
+                    logger.log("Parsing shader version failed, out of range exception: %s", system::ILogger::ELL_ERROR, e.what());
+                }
+            }
+            else
+            {
+                logger.log("Incorrect -T argument value.\nExpecting string with exactly 2 '_' delimiters: between shader stage, version major and version minor.",
+                    system::ILogger::ELL_ERROR);
+            }
+        }
+        else 
+        {
+            logger.log("invalid shader stage '%s' argument, expecting a string of length >= 6 ", system::ILogger::ELL_ERROR, s);
+        } 
```

Comment:
but you don't recover, you don't set the `stageStr` to anything before you start testing for `s.length()>=6`